--- a/Fabric/build.gradle
+++ b/Fabric/build.gradle
@@ -9,9 +9,6 @@
         exclude(group: "net.fabricmc.fabric-api")
     }
 
-    stupidRemapArch("dev.ftb.mods:ftb-essentials:${ftb_essentials}")
-    stupidRemapArch("dev.ftb.mods:ftb-ranks:${ftb_ranks}")
-
     modImplementation "maven.modrinth:fabrictailor:${fabrictailor}"
     modImplementation "maven.modrinth:vanish:${vanish}"
 
@@ -116,8 +113,8 @@
     setVersionType("release")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[FABRIC/QUILT 1.21.3] CraterLib - ${project.version}")
-    setGameVersions("1.21.3")
+    setDisplayName("[FABRIC/QUILT 1.18.2] CraterLib - ${project.version}")
+    setGameVersions("1.18.2")
     setLoaders("fabric", "quilt")
     setArtifact(remapJar)
     setCurseEnvironment("both")
