--- a/Fabric/build.gradle
+++ b/Fabric/build.gradle
@@ -15,15 +15,11 @@
         exclude(group: "net.fabricmc.fabric-api")
     }
 
-    stupidRemapArch("dev.ftb.mods:ftb-essentials:${ftb_essentials}")
-    stupidRemapArch("dev.ftb.mods:ftb-ranks:${ftb_ranks}")
-
     modImplementation("me.shedaniel.cloth:cloth-config-fabric:${cloth_config}")
     modImplementation "maven.modrinth:fabrictailor:${fabrictailor}"
     modImplementation "maven.modrinth:vanish:${vanish}"
-    modImplementation("unimaven.modrinth:advanced-chat:${advanced_chat}")
-    modImplementation("unimaven.modrinth:player-roles:1.6.15+1.21.6")
-    modImplementation("dev.gegy:player-roles-api:1.6.15")
+    modImplementation("dev.gegy:player-roles:1.5.2")
+    modImplementation("dev.gegy:player-roles-api:1.5.2")
 
     // Do not edit or remove
     implementation project(":Common")
@@ -126,8 +122,8 @@
     setVersionType("release")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[FABRIC/QUILT 1.21.9/10] CraterLib - ${project.version}")
-    setGameVersions("1.21.9", "1.21.10")
+    setDisplayName("[FABRIC/QUILT 1.18.2] CraterLib - ${project.version}")
+    setGameVersions("1.18.2")
     setLoaders("fabric", "quilt")
     setArtifact(remapJar)
     setCurseEnvironment("both")
