--- a/Common/src/main/java/com/hypherionmc/craterlib/mixin/events/ServerStatusMixin.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/mixin/events/ServerStatusMixin.java
@@ -14,13 +14,13 @@
 @Mixin(ServerStatus.class)
 public class ServerStatusMixin {
 
-    @Inject(method = "favicon", at = @At("RETURN"), cancellable = true)
-    private void injectIconEvent(CallbackInfoReturnable<Optional<ServerStatus.Favicon>> cir) {
-        ServerStatusEvent.FaviconRequestEvent event = new ServerStatusEvent.FaviconRequestEvent(cir.getReturnValue().isEmpty() ? Optional.empty() : Optional.of(new WrappedServerStatus.WrappedFavicon(cir.getReturnValue().get())));
+    @Inject(method = "getFavicon", at = @At("RETURN"), cancellable = true)
+    private void injectIconEvent(CallbackInfoReturnable<String> cir) {
+        ServerStatusEvent.FaviconRequestEvent event = new ServerStatusEvent.FaviconRequestEvent(cir.getReturnValue().isEmpty() ? Optional.empty() : Optional.of(new WrappedServerStatus.WrappedFavicon(cir.getReturnValue())));
         CraterEventBus.INSTANCE.postEvent(event);
 
         if (event.getNewIcon().isPresent()) {
-            cir.setReturnValue(Optional.of(event.getNewIcon().get().toMojang()));
+            cir.setReturnValue(event.getNewIcon().get().toMojang());
         }
     }
 
