--- a/Fabric/build.gradle
+++ b/Fabric/build.gradle
@@ -110,8 +110,8 @@
     setVersionType("release")
     setChangelog("https://raw.githubusercontent.com/hypherionmc/changelogs/main/craterlib/changelog-fabric.md")
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[FABRIC/QUILT 1.20.6] CraterLib - ${project.version}")
-    setGameVersions("1.20.6")
+    setDisplayName("[FABRIC/QUILT 1.20/1.20.1] CraterLib - ${project.version}")
+    setGameVersions("1.20", "1.20.1")
     setLoaders("fabric", "quilt")
     setArtifact(remapJar)
     setCurseEnvironment("both")
