--- a/Fabric/build.gradle
+++ b/Fabric/build.gradle
@@ -9,6 +9,7 @@
         exclude(group: "net.fabricmc.fabric-api")
     }
 
+    stupidRemapArch("dev.ftb.mods:ftb-essentials:${ftb_essentials}")
     modImplementation "maven.modrinth:fabrictailor:${fabrictailor}"
     modImplementation "maven.modrinth:vanish:${vanish}"
 
@@ -113,8 +114,8 @@
     setVersionType("release")
     setChangelog("https://raw.githubusercontent.com/hypherionmc/changelogs/main/craterlib/changelog-fabric.md")
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[FABRIC/QUILT 1.21.2] CraterLib - ${project.version}")
-    setGameVersions("1.21.2")
+    setDisplayName("[FABRIC/QUILT 1.20/1.20.1] CraterLib - ${project.version}")
+    setGameVersions("1.20", "1.20.1")
     setLoaders("fabric", "quilt")
     setArtifact(remapJar)
     setCurseEnvironment("both")
