--- a/.jenkins/Jenkinsfile.snapshot
+++ b/.jenkins/Jenkinsfile.snapshot
@@ -1,10 +1,10 @@
 def projectName = "CraterLib";
 def projectIcon = "https://cdn.modrinth.com/data/Nn8Wasaq/a172c634683a11a2e9ae593e56eba7885743bb44.png";
-def JDK = "21";
-def majorMc = "1.21";
-def modLoaders = "neoforge|fabric|quilt";
-def supportedMc = "1.21";
-def reltype = "port";
+def JDK = "17";
+def majorMc = "1.19.2";
+def modLoaders = "forge|fabric|quilt";
+def supportedMc = "1.19.2";
+def reltype = "snapshot";
 
 pipeline {
     agent {
