--- a/Common/src/main/java/com/hypherionmc/craterlib/client/gui/config/CraterConfigScreen.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/client/gui/config/CraterConfigScreen.java
@@ -14,7 +14,6 @@
 import net.minecraft.client.gui.GuiGraphics;
 import net.minecraft.client.gui.screens.ConfirmScreen;
 import net.minecraft.client.gui.screens.Screen;
-import net.minecraft.client.input.MouseButtonEvent;
 import net.minecraft.client.renderer.GameRenderer;
 import net.minecraft.network.chat.Component;
 import net.minecraft.resources.ResourceLocation;
@@ -38,7 +37,6 @@
 /**
  * @author HypherionSA
  */
-@SuppressWarnings("deprecation")
 @Deprecated(forRemoval = true, since = "2.1.3")
 public class CraterConfigScreen extends Screen {
     public static final float SCROLLBAR_BOTTOM_COLOR = .5f;
@@ -181,7 +179,7 @@
 
     @Override
     public void render(@NotNull GuiGraphics matrices, int mouseX, int mouseY, float delta) {
-        /*overlayBackground(matrices.pose(), TOP, height - BOTTOM, 32);
+        overlayBackground(matrices.pose(), TOP, height - BOTTOM, 32);
         renderScrollBar();
 
         matrices.pose().pushPose();
@@ -200,7 +198,7 @@
             option.render(minecraft, font, 40, y, width - 80, height1, matrices, mouseX, mouseY, delta);
             renderConfigTooltip(matrices, font, mouseX, mouseY, 40, y, font.width(option.text), height1, option.text.getString(), option.getLangKeys().toArray(new String[0]));
             y += height1;
-        }*/
+        }
     }
 
     private void renderScrollBar() {
@@ -235,8 +233,8 @@
             buffer.addVertex((scrollbarPositionMaxX - 1), (minY + height - 1), 0.0f).setColor(SCROLLBAR_TOP_COLOR, SCROLLBAR_TOP_COLOR, SCROLLBAR_TOP_COLOR, 1);
             buffer.addVertex((scrollbarPositionMaxX - 1), minY, 0.0f).setColor(SCROLLBAR_TOP_COLOR, SCROLLBAR_TOP_COLOR, SCROLLBAR_TOP_COLOR, 1);
             buffer.addVertex(scrollbarPositionMinX, minY, 0.0f).setColor(SCROLLBAR_TOP_COLOR, SCROLLBAR_TOP_COLOR, SCROLLBAR_TOP_COLOR, 1);
-            //BufferUploader.drawWithShader(buffer.buildOrThrow());
-            //RenderSystem.disableBlend();
+            BufferUploader.drawWithShader(buffer.buildOrThrow());
+            RenderSystem.disableBlend();
             //RenderSystem.enableTexture();
         }
     }
@@ -244,8 +242,8 @@
     private void renderShadow(PoseStack matrices) {
         Tesselator tesselator = Tesselator.getInstance();
         BufferBuilder buffer = tesselator.begin(VertexFormat.Mode.QUADS, DefaultVertexFormat.POSITION_TEX_COLOR);
-        //RenderSystem.enableBlend();
-        //RenderSystem.blendFuncSeparate(770, 771, 0, 1);
+        RenderSystem.enableBlend();
+        RenderSystem.blendFuncSeparate(770, 771, 0, 1);
         //RenderSystem.disableTexture();
         //RenderSystem.setShader(GameRenderer::getPositionTexColorShader);
         Matrix4f matrix = matrices.last().pose();
@@ -257,9 +255,9 @@
         buffer.addVertex(matrix, width, height - BOTTOM, 0.0F).setUv(1, 1).setColor(0, 0, 0, 185);
         buffer.addVertex(matrix, width, height - BOTTOM - 4, 0.0F).setUv(1, 0).setColor(0, 0, 0, 0);
         buffer.addVertex(matrix, 0, height - BOTTOM - 4, 0.0F).setUv(0, 0).setColor(0, 0, 0, 0);
-        //BufferUploader.drawWithShader(buffer.buildOrThrow());
+        BufferUploader.drawWithShader(buffer.buildOrThrow());
         //RenderSystem.enableTexture();
-        //RenderSystem.disableBlend();
+        RenderSystem.disableBlend();
     }
 
     protected void overlayBackground(PoseStack matrices, int h1, int h2, int color) {
@@ -270,13 +268,13 @@
         Tesselator tesselator = Tesselator.getInstance();
         BufferBuilder buffer = tesselator.begin(VertexFormat.Mode.QUADS, DefaultVertexFormat.POSITION_TEX_COLOR);
         //RenderSystem.setShader(GameRenderer::getPositionTexColorShader);
-        //RenderSystem.setShaderTexture(0, Screen.MENU_BACKGROUND);
-        //RenderSystem.setShaderColor(1.0f, 1.0f, 1.0f, 1.0f);
+        RenderSystem.setShaderTexture(0, Screen.MENU_BACKGROUND);
+        RenderSystem.setShaderColor(1.0f, 1.0f, 1.0f, 1.0f);
         buffer.addVertex(matrix, minX, maxY, 0.0F).setUv(minX / 32.0F, maxY / 32.0F).setColor(red, green, blue, endAlpha);
         buffer.addVertex(matrix, maxX, maxY, 0.0F).setUv(maxX / 32.0F, maxY / 32.0F).setColor(red, green, blue, endAlpha);
         buffer.addVertex(matrix, maxX, minY, 0.0F).setUv(maxX / 32.0F, minY / 32.0F).setColor(red, green, blue, startAlpha);
         buffer.addVertex(matrix, minX, minY, 0.0F).setUv(minX / 32.0F, minY / 32.0F).setColor(red, green, blue, startAlpha);
-        //BufferUploader.drawWithShader(buffer.buildOrThrow());
+        BufferUploader.drawWithShader(buffer.buildOrThrow());
     }
 
     public int scrollHeight() {
@@ -343,17 +341,17 @@
     }
 
     @Override
-    public boolean mouseClicked(MouseButtonEvent arg, boolean bl) {
-        this.dragging = arg.input() == 0 && arg.x() >= width - 6 && arg.y() < width;
-        return super.mouseClicked(arg, bl);
+    public boolean mouseClicked(double d, double e, int i) {
+        this.dragging = i == 0 && d >= width - 6 && d < width;
+        return super.mouseClicked(d, e, i) || dragging;
     }
 
     @Override
-    public boolean mouseDragged(MouseButtonEvent arg, double d, double e) {
-        if (super.mouseDragged(arg, d, e)) {
+    public boolean mouseDragged(double d, double e, int i, double f, double g) {
+        if (super.mouseDragged(d, e, i, f, g)) {
             return true;
         }
-        if (arg.input() != 0 || !this.dragging) {
+        if (i != 0 || !this.dragging) {
             return false;
         }
         if (e < TOP) {
@@ -365,7 +363,7 @@
             int j = height - BOTTOM - TOP;
             int k = Mth.clamp((int) ((float) (j * j) / (float) this.scrollHeight()), 32, j - 8);
             double l = Math.max(1.0, h / (double) (j - k));
-            //scrollerAmount = Mth.clamp(scrollerAmount + arg.scr * l, 0, scrollHeight());
+            scrollerAmount = Mth.clamp(scrollerAmount + g * l, 0, scrollHeight());
         }
         return true;
     }
@@ -386,7 +384,7 @@
                 for (String desc : description) {
                     list.add(Component.translatable(desc));
                 }
-                //stack.renderComponentTooltip(font, list, mouseX, mouseY);
+                stack.renderComponentTooltip(font, list, mouseX, mouseY);
             }
         }
     }
