--- a/Common/src/main/java/com/hypherionmc/craterlib/nojang/commands/BridgedCommandSourceStack.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/nojang/commands/BridgedCommandSourceStack.java
@@ -19,7 +19,7 @@
         if (!internal.getServer().getGameRules().getBoolean(RULE_SENDCOMMANDFEEDBACK)) {
             internal.sendSystemMessage(ChatUtils.adventureToMojang(supplier.get()));
         } else {
-            internal.sendSuccess(() -> ChatUtils.adventureToMojang(supplier.get()), bl);
+            internal.sendSuccess(ChatUtils.adventureToMojang(supplier.get()), bl);
         }
     }
 
