--- a/Common/src/main/java/com/hypherionmc/craterlib/client/gui/config/widgets/InternalConfigButton.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/client/gui/config/widgets/InternalConfigButton.java
@@ -1,12 +1,12 @@
 package com.hypherionmc.craterlib.client.gui.config.widgets;
 
 import com.hypherionmc.craterlib.client.gui.config.CraterConfigScreen;
-import net.minecraft.client.gui.GuiGraphics;
+import com.mojang.blaze3d.vertex.PoseStack;
 import net.minecraft.client.gui.components.AbstractButton;
 import net.minecraft.client.gui.narration.NarratedElementType;
 import net.minecraft.client.gui.narration.NarrationElementOutput;
-import net.minecraft.client.input.InputWithModifiers;
 import net.minecraft.network.chat.Component;
+import org.jetbrains.annotations.NotNull;
 
 /**
  * @author HypherionSA
@@ -24,7 +24,7 @@
     }
 
     @Override
-    protected void renderWidget(GuiGraphics arg, int i, int j, float f) {
+    public void render(@NotNull PoseStack poseStack, int i, int j, float f) {
         if (cancel) {
             setMessage(Component.translatable(screen.isEdited() ? "t.clc.cancel_discard" : "gui.cancel"));
         } else {
@@ -32,7 +32,7 @@
             active = screen.isEdited() && !hasErrors;
             setMessage(Component.translatable(hasErrors ? "t.clc.error" : "t.clc.save"));
         }
-        super.renderWidget(arg, i, j, f);
+        super.render(poseStack, i, j, f);
     }
 
     @Override
@@ -41,12 +41,13 @@
     }
 
     @Override
-    public void onPress(InputWithModifiers arg) {
+    public void onPress() {
         if (cancel) {
             screen.onClose();
         } else {
             screen.save();
         }
     }
+
 
 }
