--- a/NeoForge/build.gradle
+++ b/NeoForge/build.gradle
@@ -39,12 +39,7 @@
 
 unimined.minecraft {
     neoForged {
-        if (isPort) {
-            loader "net.neoforged:neoforge:${neoforge_version}:universal"
-        } else {
-            loader neoforge_version
-        }
-
+        loader neoforge_version
         mixinConfig("${mod_id}.mixins.json", "${mod_id}.neoforge.mixins.json")
     }
 }
@@ -118,8 +113,8 @@
     setVersionType("release")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[NeoForge 1.21.9/10] CraterLib - ${project.version}")
-    setGameVersions("1.21.9", "1.21.10")
+    setDisplayName("[NeoForge 1.21.3/1.21.4] CraterLib - ${project.version}")
+    setGameVersions("1.21.3", "1.21.4")
     setLoaders("neoforge")
     setArtifact(remapJar)
     setCurseEnvironment("both")
