--- a/build.gradle
+++ b/build.gradle
@@ -29,8 +29,8 @@
     apply plugin: 'com.github.johnrengelman.shadow'
     apply plugin: 'com.hypherionmc.modutils.modpublisher'
 
-    sourceCompatibility = JavaVersion.VERSION_21
-    targetCompatibility = JavaVersion.VERSION_21
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
 
     group = rootProject.group
 
@@ -66,8 +66,6 @@
         shade "net.kyori:adventure-text-serializer-gson:${adventure}"
         shade "net.kyori:adventure-text-minimessage:${adventure}"
 
-        stupidRemapArch("dev.ftb.mods:ftb-essentials:${ftb_essentials}")
-
         compileOnly 'net.luckperms:api:5.4'
         compileOnly("org.projectlombok:lombok:${lombok}")
         annotationProcessor("org.projectlombok:lombok:${lombok}")
@@ -112,7 +110,7 @@
 
     tasks.withType(JavaCompile).configureEach {
         it.options.encoding = 'UTF-8'
-        it.options.release = 21
+        it.options.release = 17
     }
 
     tasks.withType(GenerateModuleMetadata).configureEach {
