--- a/build.gradle
+++ b/build.gradle
@@ -29,8 +29,8 @@
     apply plugin: 'com.github.johnrengelman.shadow'
     apply plugin: 'com.hypherionmc.modutils.modpublisher'
 
-    sourceCompatibility = JavaVersion.VERSION_21
-    targetCompatibility = JavaVersion.VERSION_21
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
 
     group = rootProject.group
 
