--- a/NeoForge/build.gradle
+++ b/NeoForge/build.gradle
@@ -34,12 +34,7 @@
 
 unimined.minecraft {
     neoForged {
-        if (isPort) {
-            loader "net.neoforged:neoforge:${neoforge_version}:universal"
-        } else {
-            loader neoforge_version
-        }
-
+        loader neoforge_version
         mixinConfig("${mod_id}.mixins.json", "${mod_id}.neoforge.mixins.json")
     }
 }
@@ -113,8 +108,8 @@
     setVersionType("release")
     setChangelog("https://raw.githubusercontent.com/hypherionmc/changelogs/main/craterlib/changelog-forge.md")
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[NeoForge 1.21.2] CraterLib - ${project.version}")
-    setGameVersions("1.21.2")
+    setDisplayName("[NeoForge 1.21.x] CraterLib - ${project.version}")
+    setGameVersions("1.21", "1.21.1")
     setLoaders("neoforge")
     setArtifact(remapJar)
     setCurseEnvironment("both")
