--- a/Common/src/main/java/com/hypherionmc/craterlib/nojang/server/BridgedMinecraftServer.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/nojang/server/BridgedMinecraftServer.java
@@ -38,7 +38,7 @@
     }
 
     public String getName() {
-        return SharedConstants.getCurrentVersion().name();
+        return SharedConstants.getCurrentVersion().getName();
     }
 
     public boolean usesAuthentication() {
@@ -50,21 +50,21 @@
     }
 
     public boolean isPlayerBanned(BridgedGameProfile profile) {
-        return internal.getPlayerList().getBans().isBanned(profile.toNameAndId());
+        return internal.getPlayerList().getBans().isBanned(profile.toMojang());
     }
 
     public void whitelistPlayer(BridgedGameProfile gameProfile) {
         if (!internal.getPlayerList().isUsingWhitelist())
             return;
 
-        internal.getPlayerList().getWhiteList().add(new UserWhiteListEntry(gameProfile.toNameAndId()));
+        internal.getPlayerList().getWhiteList().add(new UserWhiteListEntry(gameProfile.toMojang()));
     }
 
     public void unWhitelistPlayer(BridgedGameProfile gameProfile) {
         if (!internal.getPlayerList().isUsingWhitelist())
             return;
 
-        internal.getPlayerList().getWhiteList().remove(new UserWhiteListEntry(gameProfile.toNameAndId()));
+        internal.getPlayerList().getWhiteList().remove(new UserWhiteListEntry(gameProfile.toMojang()));
     }
 
     public List<BridgedPlayer> getPlayers() {
@@ -83,7 +83,7 @@
     }
 
     public void banPlayer(BridgedGameProfile profile) {
-        internal.getPlayerList().getBans().add(new UserBanListEntry(profile.toNameAndId()));
+        internal.getPlayerList().getBans().add(new UserBanListEntry(profile.toMojang()));
     }
 
     public void executeCommand(BridgedMinecraftServer server, BridgedFakePlayer player, String command) {
