--- a/Forge/src/main/java/com/hypherionmc/craterlib/CraterLib.java
+++ b/Forge/src/main/java/com/hypherionmc/craterlib/CraterLib.java
@@ -2,6 +2,7 @@
 
 import com.hypherionmc.craterlib.api.events.client.LateInitEvent;
 import com.hypherionmc.craterlib.common.ForgeServerEvents;
+import com.hypherionmc.craterlib.compat.Vanish;
 import com.hypherionmc.craterlib.core.event.CraterEventBus;
 import com.hypherionmc.craterlib.core.networking.CraterPacketNetwork;
 import com.hypherionmc.craterlib.core.networking.data.PacketSide;
@@ -32,5 +33,9 @@
             LateInitEvent event = new LateInitEvent(new BridgedMinecraft(), BridgedOptions.of(Minecraft.getInstance().options));
             CraterEventBus.INSTANCE.postEvent(event);
         });
+
+        if (ModloaderEnvironment.INSTANCE.isModLoaded("vmod")) {
+            MinecraftForge.EVENT_BUS.register(new Vanish());
+        }
     }
 }
