--- a/Forge/build.gradle
+++ b/Forge/build.gradle
@@ -3,11 +3,16 @@
 
 dependencies {
     // Compat
-    // NOT AVAILABLE ON FORGE modImplementation("maven.modrinth:vanishmod:${vanishmod}")
+    modImplementation("maven.modrinth:vanishmod:${vanishmod}")
+
+    stupidRemapArch("dev.ftb.mods:ftb-essentials-forge:${ftb_essentials}")
+    stupidRemapArch("dev.ftb.mods:ftb-ranks-forge:${ftb_ranks}")
 
     modImplementation("me.shedaniel.cloth:cloth-config-forge:${cloth_config}")
-    modImplementation("unimaven.curseforge:playerrevive-266890:${player_revive}")
-    modImplementation("unimaven.curseforge:creativecore-257814:${creative_core}")
+
+    // NOT AVAILABLE FOR FORGE
+    //modImplementation("unimaven.curseforge:playerrevive-266890:5019850")
+    //modImplementation("unimaven.curseforge:creativecore-257814:5140174")
 
     // Do not edit or remove
     implementation project(":Common")
@@ -111,8 +116,8 @@
     setVersionType("release")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[Forge 1.20.6] CraterLib - ${project.version}")
-    setGameVersions("1.20.6")
+    setDisplayName("[Forge 1.20.4] CraterLib - ${project.version}")
+    setGameVersions("1.20.4")
     setLoaders("forge")
     setArtifact(remapJar)
     setCurseEnvironment("both")
