--- a/build.gradle
+++ b/build.gradle
@@ -11,7 +11,6 @@
     multiProject = true
     enableMirrorMaven = true
     enableReleasesMaven = true
-
     dopplerToken = System.getenv("DOPPLER_KEY")
 
     versioning {
@@ -29,8 +28,8 @@
     apply plugin: 'com.github.johnrengelman.shadow'
     apply plugin: 'com.hypherionmc.modutils.modpublisher'
 
-    sourceCompatibility = JavaVersion.VERSION_21
-    targetCompatibility = JavaVersion.VERSION_21
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
 
     group = rootProject.group
 
@@ -61,13 +60,11 @@
         shade "me.hypherionmc.moon-config:toml:${moon_config}"
         shade "me.hypherionmc.moon-config:json:${moon_config}"
         shade "com.hypherionmc:rpcsdk:${rpc_sdk}"
-        shade "me.hypherionmc.sdlink:mcdiscordformatter-1.20.3:${discord_formatter}"
+        shade "me.hypherionmc.sdlink:mcdiscordformatter-1.19.1:${discord_formatter}"
         shade "net.kyori:adventure-api:${adventure}"
         shade "net.kyori:adventure-text-serializer-gson:${adventure}"
         shade "net.kyori:adventure-text-minimessage:${adventure}"
 
-        stupidRemapArch("dev.ftb.mods:ftb-essentials:${ftb_essentials}")
-
         compileOnly 'net.luckperms:api:5.4'
         compileOnly("org.projectlombok:lombok:${lombok}")
         annotationProcessor("org.projectlombok:lombok:${lombok}")
@@ -112,7 +109,7 @@
 
     tasks.withType(JavaCompile).configureEach {
         it.options.encoding = 'UTF-8'
-        it.options.release = 21
+        it.options.release = 17
     }
 
     tasks.withType(GenerateModuleMetadata).configureEach {
