--- a/build.gradle
+++ b/build.gradle
@@ -7,21 +7,14 @@
     id 'maven-publish'
 }
 
-var relType = project.properties["releaseType"] ?: "${release_type}"
-
-ext {
-    isPort = (relType == 'port')
-}
-
 orion.setup {
     multiProject = true
     enableMirrorMaven = true
     enableReleasesMaven = true
-    enableSnapshotsMaven = isPort
-
     dopplerToken = System.getenv("DOPPLER_KEY")
 
     versioning {
+        var relType = project.properties["releaseType"] ?: "${release_type}"
         identifier("${relType}")
     }
 }
@@ -35,14 +28,13 @@
     apply plugin: 'com.github.johnrengelman.shadow'
     apply plugin: 'com.hypherionmc.modutils.modpublisher'
 
-    sourceCompatibility = JavaVersion.VERSION_21
-    targetCompatibility = JavaVersion.VERSION_21
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
 
     group = rootProject.group
 
     repositories {
         mavenCentral()
-        mavenLocal()
 
         maven {
             name = "Modrinth"
@@ -68,7 +60,7 @@
         shade "me.hypherionmc.moon-config:toml:${moon_config}"
         shade "me.hypherionmc.moon-config:json:${moon_config}"
         shade "com.hypherionmc:rpcsdk:${rpc_sdk}"
-        shade "me.hypherionmc.sdlink:mcdiscordformatter-1.20.3:${discord_formatter}"
+        shade "me.hypherionmc.sdlink:mcdiscordformatter-1.19.1:${discord_formatter}"
         shade "net.kyori:adventure-api:${adventure}"
         shade "net.kyori:adventure-text-serializer-gson:${adventure}"
         shade "net.kyori:adventure-text-minimessage:${adventure}"
@@ -117,7 +109,7 @@
 
     tasks.withType(JavaCompile).configureEach {
         it.options.encoding = 'UTF-8'
-        it.options.release = 21
+        it.options.release = 17
     }
 
     tasks.withType(GenerateModuleMetadata).configureEach {
