--- a/Paper/src/main/java/com/hypherionmc/craterlib/common/PaperCompatHelper.java
+++ b/Paper/src/main/java/com/hypherionmc/craterlib/common/PaperCompatHelper.java
@@ -4,7 +4,6 @@
 import com.hypherionmc.craterlib.core.platform.ModloaderEnvironment;
 import com.hypherionmc.craterlib.nojang.world.entity.player.BridgedPlayer;
 import com.hypherionmc.craterlib.utils.ChatUtils;
-import net.kyori.adventure.text.Component;
 import org.bukkit.Bukkit;
 import org.bukkit.entity.Player;
 import org.bukkit.metadata.MetadataValue;
