--- a/Common/src/main/java/com/hypherionmc/craterlib/nojang/commands/BridgedCommandSourceStack.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/nojang/commands/BridgedCommandSourceStack.java
@@ -1,5 +1,6 @@
 package com.hypherionmc.craterlib.nojang.commands;
 
+import com.hypherionmc.craterlib.CraterConstants;
 import com.hypherionmc.craterlib.nojang.world.entity.player.BridgedPlayer;
 import com.hypherionmc.craterlib.utils.ChatUtils;
 import lombok.RequiredArgsConstructor;
@@ -14,7 +15,7 @@
     private final CommandSourceStack internal;
 
     public void sendSuccess(Supplier<Component> supplier, boolean bl) {
-        internal.sendSuccess(() -> ChatUtils.adventureToMojang(supplier.get()), bl);
+        internal.sendSuccess(ChatUtils.adventureToMojang(supplier.get()), bl);
     }
 
     public void sendFailure(Component text) {
@@ -22,11 +23,22 @@
     }
 
     public boolean isPlayer() {
-        return internal.isPlayer();
+        try {
+            internal.getPlayerOrException();
+            return true;
+        } catch (Exception ignored) {
+            return false;
+        }
     }
 
     public BridgedPlayer getPlayer() {
-        return BridgedPlayer.of(internal.getPlayer());
+        try {
+            return BridgedPlayer.of(internal.getPlayerOrException());
+        } catch (Exception e) {
+            CraterConstants.LOG.error("Failed to retrieve player", e);
+        }
+
+        return null;
     }
 
     public CommandSourceStack toMojang() {
