--- a/Common/src/main/java/com/hypherionmc/craterlib/nojang/commands/BridgedCommandSourceStack.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/nojang/commands/BridgedCommandSourceStack.java
@@ -1,5 +1,6 @@
 package com.hypherionmc.craterlib.nojang.commands;
 
+import com.hypherionmc.craterlib.CraterConstants;
 import com.hypherionmc.craterlib.nojang.world.entity.player.BridgedPlayer;
 import com.hypherionmc.craterlib.utils.ChatUtils;
 import lombok.RequiredArgsConstructor;
@@ -17,14 +18,18 @@
 
     public void sendSuccess(Supplier<Component> supplier, boolean bl) {
         if (!internal.getServer().getGameRules().getBoolean(RULE_SENDCOMMANDFEEDBACK)) {
-            internal.sendSystemMessage(ChatUtils.adventureToMojang(supplier.get()));
+            try {
+                internal.getPlayerOrException().displayClientMessage(ChatUtils.adventureToMojang(supplier.get()), false);
+            } catch (Exception ignored) {}
         } else {
-            internal.sendSuccess(() -> ChatUtils.adventureToMojang(supplier.get()), bl);
+            internal.sendSuccess(ChatUtils.adventureToMojang(supplier.get()), bl);
         }
     }
 
     public void sendMessage(Component text) {
-        internal.sendSystemMessage(ChatUtils.adventureToMojang(text));
+        try {
+            internal.getPlayerOrException().displayClientMessage(ChatUtils.adventureToMojang(text), false);
+        } catch (Exception ignored) {}
     }
 
     public void sendFailure(Component text) {
@@ -32,11 +37,22 @@
     }
 
     public boolean isPlayer() {
-        return internal.isPlayer();
+        try {
+            internal.getPlayerOrException();
+            return true;
+        } catch (Exception ignored) {
+            return false;
+        }
     }
 
     public BridgedPlayer getPlayer() {
-        return BridgedPlayer.of(internal.getPlayer());
+        try {
+            return BridgedPlayer.of(internal.getPlayerOrException());
+        } catch (Exception e) {
+            CraterConstants.LOG.error("Failed to retrieve player", e);
+        }
+
+        return null;
     }
 
     public CommandSourceStack toMojang() {
