--- a/Common/src/main/java/com/hypherionmc/craterlib/nojang/commands/CommandsRegistry.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/nojang/commands/CommandsRegistry.java
@@ -70,7 +70,7 @@
                     profiles.forEach(p -> bridgedGameProfiles.add(BridgedGameProfile.of(p)));
 
                     ((TriConsumer<BridgedPlayer, List<BridgedGameProfile>, BridgedCommandSourceStack>) pair.getRight())
-                            .accept(BridgedPlayer.of(context.getSource().getPlayer()), bridgedGameProfiles, BridgedCommandSourceStack.of(context.getSource()));
+                            .accept(BridgedPlayer.of(context.getSource().getPlayerOrException()), bridgedGameProfiles, BridgedCommandSourceStack.of(context.getSource()));
                     return 1;
                 }
 
