--- a/Common/src/main/java/com/hypherionmc/craterlib/utils/ChatUtils.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/utils/ChatUtils.java
@@ -1,11 +1,6 @@
 package com.hypherionmc.craterlib.utils;
 
-import com.google.gson.JsonElement;
-import com.google.gson.JsonParseException;
-import com.hypherionmc.craterlib.core.platform.CommonPlatform;
-import com.hypherionmc.craterlib.core.platform.ModloaderEnvironment;
 import com.hypherionmc.craterlib.nojang.resources.ResourceIdentifier;
-import com.mojang.serialization.JsonOps;
 import lombok.Getter;
 import me.hypherionmc.mcdiscordformatter.discord.DiscordSerializer;
 import me.hypherionmc.mcdiscordformatter.minecraft.MinecraftSerializer;
@@ -16,57 +11,30 @@
 import net.minecraft.ChatFormatting;
 import net.minecraft.SharedConstants;
 import net.minecraft.Util;
-// @noplugin
-import net.minecraft.client.Minecraft;
-// #noplugin
-import net.minecraft.core.HolderLookup;
-import net.minecraft.core.RegistryAccess;
 import net.minecraft.network.chat.Component;
-import net.minecraft.network.chat.ComponentSerialization;
 import net.minecraft.network.chat.Style;
-import net.minecraft.util.StrictJsonParser;
 
 public class ChatUtils {
 
     @Getter
     private static final GsonComponentSerializer adventureSerializer = GsonComponentSerializer.builder().options(
-            JSONOptions.byDataVersion().at(SharedConstants.getCurrentVersion().dataVersion().version())
+            JSONOptions.byDataVersion().at(SharedConstants.getCurrentVersion().getDataVersion().getVersion())
     ).build();
 
     private static final MiniMessage miniMessage = MiniMessage.miniMessage();
 
     public static Component adventureToMojang(net.kyori.adventure.text.Component inComponent) {
-        final JsonElement serialised = adventureSerializer.serializeToTree(inComponent);
-
-        // FUCK YOU MOJANG. SERIOUSLY. FUCK OFF WITH THIS SHIT
-        return ComponentSerialization.CODEC
-                .parse(getRegistryLookup().createSerializationContext(JsonOps.INSTANCE), serialised)
-                .getOrThrow(JsonParseException::new);
+        final String serialised = adventureSerializer.serialize(inComponent);
+        return Component.Serializer.fromJson(serialised);
     }
 
     public static net.kyori.adventure.text.Component mojangToAdventure(Component inComponent) {
         try {
-            // FUCK YOU MOJANG. SERIOUSLY. FUCK OFF WITH THIS SHIT
-            final JsonElement serialised = ComponentSerialization.CODEC
-                    .encodeStart(JsonOps.INSTANCE, inComponent)
-                    .getOrThrow(JsonParseException::new);
-
-            return adventureSerializer.deserializeFromTree(serialised);
+        final String serialised = Component.Serializer.toJson(inComponent);
+            return adventureSerializer.deserialize(serialised);
         } catch (Exception e) {
             return net.kyori.adventure.text.Component.text(inComponent.getString());
         }
-    }
-
-    private static HolderLookup.Provider getRegistryLookup() {
-        // @noplugin
-        if (ModloaderEnvironment.INSTANCE.getEnvironment().isClient() && Minecraft.getInstance().level != null)
-            return Minecraft.getInstance().level.registryAccess();
-        // #noplugin
-
-        if (ModloaderEnvironment.INSTANCE.getEnvironment().isServer() && CommonPlatform.INSTANCE.getMCServer() != null)
-            return CommonPlatform.INSTANCE.getMCServer().toMojang().registryAccess();
-
-        return RegistryAccess.EMPTY;
     }
 
     // Some text components contain duplicate text, resulting in duplicate messages
