--- a/Common/src/main/java/com/hypherionmc/craterlib/nojang/resources/ResourceIdentifier.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/nojang/resources/ResourceIdentifier.java
@@ -11,7 +11,7 @@
     }
 
     public ResourceIdentifier(String path) {
-        this.internal = ResourceLocation.withDefaultNamespace(path);
+        this.internal = ResourceLocation.parse(path);
     }
 
     public String getNamespace() {
