--- a/Common/src/main/java/com/hypherionmc/craterlib/client/gui/config/widgets/SubConfigWidget.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/client/gui/config/widgets/SubConfigWidget.java
@@ -2,12 +2,12 @@
 
 import com.hypherionmc.craterlib.client.gui.config.CraterConfigScreen;
 import com.hypherionmc.craterlib.core.config.ModuleConfig;
+import com.mojang.blaze3d.vertex.PoseStack;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.Font;
-import net.minecraft.client.gui.GuiGraphics;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.screens.Screen;
-import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
 
 /**
  * @author HypherionSA
@@ -23,12 +23,12 @@
         this.subConfig = subConfig;
         this.screen = screen;
 
-        this.widget = addChild(Button.builder(Component.translatable("t.clc.opensubconfig"), this::openSubConfig).size(200, buttonHeight).build());
+        this.widget = addChild(new Button(0, 0, 200, buttonHeight, new TranslatableComponent("t.clc.opensubconfig"), this::openSubConfig));
     }
 
     @Override
-    public void render(Minecraft minecraft, Font font, int x, int y, int width, int height, GuiGraphics matrices, int mouseX, int mouseY, float delta) {
-        this.text = Component.literal(subConfig.getClass().getSimpleName().toLowerCase());
+    public void render(Minecraft minecraft, Font font, int x, int y, int width, int height, PoseStack matrices, int mouseX, int mouseY, float delta) {
+        this.text = new TranslatableComponent(subConfig.getClass().getSimpleName().toLowerCase());
         this.hideReset();
         super.render(minecraft, font, x, y, width, height, matrices, mouseX, mouseY, delta);
     }
