--- a/Fabric/build.gradle
+++ b/Fabric/build.gradle
@@ -11,7 +11,6 @@
 
     stupidRemapArch("dev.ftb.mods:ftb-essentials:${ftb_essentials}")
     stupidRemapArch("dev.ftb.mods:ftb-ranks:${ftb_ranks}")
-
     modImplementation("me.shedaniel.cloth:cloth-config-fabric:${cloth_config}")
     modImplementation "maven.modrinth:fabrictailor:${fabrictailor}"
     modImplementation "maven.modrinth:vanish:${vanish}"
@@ -118,8 +117,8 @@
     setVersionType("release")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[FABRIC/QUILT 1.21.6] CraterLib - ${project.version}")
-    setGameVersions("1.21.6")
+    setDisplayName("[FABRIC/QUILT 1.19.4] CraterLib - ${project.version}")
+    setGameVersions("1.19.4")
     setLoaders("fabric", "quilt")
     setArtifact(remapJar)
     setCurseEnvironment("both")
