--- a/NeoForge/build.gradle
+++ b/NeoForge/build.gradle
@@ -118,8 +118,8 @@
     setVersionType("release")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[NeoForge 1.21.6] CraterLib - ${project.version}")
-    setGameVersions("1.21.6")
+    setDisplayName("[NeoForge 1.21.5] CraterLib - ${project.version}")
+    setGameVersions("1.21.5")
     setLoaders("neoforge")
     setArtifact(remapJar)
     setCurseEnvironment("both")
