--- a/Fabric/build.gradle
+++ b/Fabric/build.gradle
@@ -17,13 +17,11 @@
 
     stupidRemapArch("dev.ftb.mods:ftb-essentials:${ftb_essentials}")
     stupidRemapArch("dev.ftb.mods:ftb-ranks:${ftb_ranks}")
-
     modImplementation("me.shedaniel.cloth:cloth-config-fabric:${cloth_config}")
     modImplementation "maven.modrinth:fabrictailor:${fabrictailor}"
     modImplementation "maven.modrinth:vanish:${vanish}"
-    modImplementation("unimaven.modrinth:advanced-chat:${advanced_chat}")
-    modImplementation("unimaven.modrinth:player-roles:1.6.15+1.21.6")
-    modImplementation("dev.gegy:player-roles-api:1.6.15")
+    modImplementation("unimaven.modrinth:player-roles:1.6.1")
+    modImplementation("dev.gegy:player-roles-api:1.6.1")
 
     // Do not edit or remove
     implementation project(":Common")
@@ -126,8 +124,8 @@
     setVersionType("release")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[FABRIC/QUILT 1.21.9] CraterLib - ${project.version}")
-    setGameVersions("1.21.9")
+    setDisplayName("[FABRIC/QUILT 1.19.2] CraterLib - ${project.version}")
+    setGameVersions("1.19.2")
     setLoaders("fabric", "quilt")
     setArtifact(remapJar)
     setCurseEnvironment("both")
