--- a/Common/src/main/java/com/hypherionmc/craterlib/nojang/client/BridgedMinecraft.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/nojang/client/BridgedMinecraft.java
@@ -39,7 +39,7 @@
     }
 
     public boolean isRealmServer() {
-        return internal.getCurrentServer() != null && internal.getCurrentServer().isRealm();
+        return internal.getCurrentServer() != null && internal.isConnectedToRealms();
     }
 
     public boolean isSinglePlayer() {
@@ -55,7 +55,7 @@
     }
 
     public String getGameVersion() {
-        return SharedConstants.getCurrentVersion().name();
+        return SharedConstants.getCurrentVersion().getName();
     }
 
     public String getUserName() {
