--- a/Fabric/build.gradle
+++ b/Fabric/build.gradle
@@ -126,8 +126,8 @@
     setVersionType("release")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[FABRIC/QUILT 1.21.9] CraterLib - ${project.version}")
-    setGameVersions("1.21.9")
+    setDisplayName("[FABRIC/QUILT 1.21.6-8] CraterLib - ${project.version}")
+    setGameVersions("1.21.6", "1.21.7", "1.21.8")
     setLoaders("fabric", "quilt")
     setArtifact(remapJar)
     setCurseEnvironment("both")
