--- a/Common/src/main/java/com/hypherionmc/craterlib/utils/ChatUtils.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/utils/ChatUtils.java
@@ -9,11 +9,10 @@
 import net.minecraft.ChatFormatting;
 import net.minecraft.SharedConstants;
 import net.minecraft.Util;
-import net.minecraft.core.RegistryAccess;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.Style;
-
-import java.util.function.Consumer;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.network.chat.TranslatableComponent;
 
 public class ChatUtils {
 
@@ -23,11 +22,11 @@
 
     public static Component adventureToMojang(net.kyori.adventure.text.Component inComponent) {
         final String serialised = adventureSerializer.serialize(inComponent);
-        return Component.Serializer.fromJson(serialised, RegistryAccess.EMPTY);
+        return Component.Serializer.fromJson(serialised);
     }
 
     public static net.kyori.adventure.text.Component mojangToAdventure(Component inComponent) {
-        final String serialised = Component.Serializer.toJson(inComponent, RegistryAccess.EMPTY);
+        final String serialised = Component.Serializer.toJson(inComponent);
         return adventureSerializer.deserialize(serialised);
     }
 
@@ -36,7 +35,7 @@
     public static Component safeCopy(Component inComponent) {
         String value = inComponent.getString();
         Style style = inComponent.getStyle();
-        return Component.literal(value).withStyle(style);
+        return new TextComponent(value).withStyle(style);
     }
 
     public static String strip(String inString, String... toStrip) {
@@ -65,7 +64,7 @@
     }
 
     public static net.kyori.adventure.text.Component resolve(String component, boolean formatted) {
-        Component returnVal = Component.literal(component);
+        Component returnVal = new TextComponent(component);
         if (formatted) {
             returnVal = MinecraftSerializer.INSTANCE.serialize(component);
         }
@@ -93,7 +92,7 @@
         if (identifier == null)
             return net.kyori.adventure.text.Component.text("Unknown");
 
-        return mojangToAdventure(Component.translatable(Util.makeDescriptionId("biome", identifier.toMojang())));
+        return mojangToAdventure(new TranslatableComponent(Util.makeDescriptionId("biome", identifier.toMojang())));
     }
 
 }
