--- a/Common/src/main/java/com/hypherionmc/craterlib/compat/FTBEssentials.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/compat/FTBEssentials.java
@@ -8,8 +8,8 @@
 public class FTBEssentials {
 
     public static boolean isPlayerMuted(BridgedPlayer player) {
-        Optional<FTBEPlayerData> data = FTBEPlayerData.getOrCreate(player.toMojang());
-        return data.map(FTBEPlayerData::isMuted).orElse(false);
+        FTBEPlayerData data = FTBEPlayerData.get(player.toMojang());
+        return data != null && data.muted;
 
     }
 
