--- a/Paper/build.gradle
+++ b/Paper/build.gradle
@@ -70,8 +70,8 @@
     setVersionType("alpha")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[Paper 1.21.3] CraterLib - ${project.version}")
-    setGameVersions("1.21.3")
+    setDisplayName("[Paper 1.19.4] CraterLib - ${project.version}")
+    setGameVersions("1.19.4")
     setLoaders("paper")
     setArtifact(reobfJar.outputJar)
 }
