--- a/Common/src/main/java/com/hypherionmc/craterlib/nojang/client/multiplayer/BridgedServerData.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/nojang/client/multiplayer/BridgedServerData.java
@@ -23,7 +23,7 @@
     }
 
     public int getMaxPlayers() {
-        if (internal.players == null) {
+        if (!internal.pinged || internal.players == null) {
             return internal.playerList.size() + 1;
         }
 
