--- a/build.gradle
+++ b/build.gradle
@@ -7,21 +7,15 @@
     id 'maven-publish'
 }
 
-var relType = project.properties["releaseType"] ?: "${release_type}"
-
-ext {
-    isPort = (relType == 'port')
-}
-
 orion.setup {
     multiProject = true
     enableMirrorMaven = true
     enableReleasesMaven = true
-    //enableSnapshotsMaven = isPort
 
     dopplerToken = System.getenv("DOPPLER_KEY")
 
     versioning {
+        var relType = project.properties["releaseType"] ?: "${release_type}"
         identifier("${relType}")
     }
 }
@@ -42,7 +36,6 @@
 
     repositories {
         mavenCentral()
-        mavenLocal()
 
         maven {
             name = "Modrinth"
