--- a/Forge/build.gradle
+++ b/Forge/build.gradle
@@ -3,7 +3,9 @@
 
 dependencies {
     // Compat
-    // NOT AVAILABLE ON FORGE modImplementation("maven.modrinth:vanishmod:${vanishmod}")
+    modImplementation("maven.modrinth:vanishmod:${vanishmod}")
+
+    stupidRemapArch("dev.ftb.mods:ftb-essentials-forge:${ftb_essentials}")
 
     // Do not edit or remove
     implementation project(":Common")
@@ -107,8 +109,8 @@
     setVersionType("release")
     setChangelog("https://raw.githubusercontent.com/hypherionmc/changelogs/main/craterlib/changelog-forge.md")
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[Forge 1.20.6] CraterLib - ${project.version}")
-    setGameVersions("1.20.6")
+    setDisplayName("[Forge 1.19.4] CraterLib - ${project.version}")
+    setGameVersions("1.19.4")
     setLoaders("forge")
     setArtifact(remapJar)
     setCurseEnvironment("both")
