--- a/Fabric/build.gradle
+++ b/Fabric/build.gradle
@@ -113,8 +113,8 @@
     setVersionType("release")
     setChangelog("https://raw.githubusercontent.com/hypherionmc/changelogs/main/craterlib/changelog-fabric.md")
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[FABRIC/QUILT 1.21.2] CraterLib - ${project.version}")
-    setGameVersions("1.21.2")
+    setDisplayName("[FABRIC/QUILT 1.21.x] CraterLib - ${project.version}")
+    setGameVersions("1.21", "1.21.1")
     setLoaders("fabric", "quilt")
     setArtifact(remapJar)
     setCurseEnvironment("both")
