--- a/Fabric/build.gradle
+++ b/Fabric/build.gradle
@@ -118,8 +118,8 @@
     setVersionType("release")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[FABRIC/QUILT 1.21.9] CraterLib - ${project.version}")
-    setGameVersions("1.21.9")
+    setDisplayName("[FABRIC/QUILT 1.21.x] CraterLib - ${project.version}")
+    setGameVersions("1.21", "1.21.1")
     setLoaders("fabric", "quilt")
     setArtifact(remapJar)
     setCurseEnvironment("both")
