--- a/Fabric/src/main/java/com/hypherionmc/craterlib/CraterLibInitializer.java
+++ b/Fabric/src/main/java/com/hypherionmc/craterlib/CraterLibInitializer.java
@@ -12,7 +12,7 @@
 import com.hypherionmc.craterlib.nojang.commands.CommandsRegistry;
 import com.hypherionmc.craterlib.nojang.server.BridgedMinecraftServer;
 import net.fabricmc.api.ModInitializer;
-import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback;
+import net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback;
 import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
 
 public class CraterLibInitializer implements ModInitializer {
@@ -20,7 +20,7 @@
     @Override
     public void onInitialize() {
         new CraterPacketNetwork(new CraterFabricNetworkHandler(PacketSide.SERVER));
-        CommandRegistrationCallback.EVENT.register((dispatcher, registryAccess, environment) -> {
+        CommandRegistrationCallback.EVENT.register((dispatcher, registryAccess) -> {
             CraterEventBus.INSTANCE.postEvent(new CraterRegisterCommandEvent());
             CommandsRegistry.INSTANCE.registerCommands(dispatcher);
         });
