--- a/.jenkins/Jenkinsfile.deploy
+++ b/.jenkins/Jenkinsfile.deploy
@@ -1,7 +1,7 @@
 pipeline {
     agent {
         docker {
-            image "registry.firstdark.dev/java21:latest"
+            image "registry.firstdark.dev/java17:latest"
             alwaysPull true
         }
     }
@@ -10,7 +10,7 @@
         stage("Notify Discord") {
             steps {
                 discordSend webhookURL: env.FDD_WH_ADMIN,
-                        title: "Deploy Started: CraterLib Port Deploy #${BUILD_NUMBER}",
+                        title: "Deploy Started: CraterLib 1.20.4 Deploy #${BUILD_NUMBER}",
                         link: env.BUILD_URL,
                         result: 'SUCCESS',
                         description: "Build: [${BUILD_NUMBER}](${env.BUILD_URL})"
@@ -39,7 +39,7 @@
             deleteDir()
 
             discordSend webhookURL: env.FDD_WH_ADMIN,
-                    title: "CraterLib Port Deploy #${BUILD_NUMBER}",
+                    title: "CraterLib 1.20.4 Deploy #${BUILD_NUMBER}",
                     link: env.BUILD_URL,
                     result: currentBuild.currentResult,
                     description: "Build: [${BUILD_NUMBER}](${env.BUILD_URL})\nStatus: ${currentBuild.currentResult}"
