--- a/Common/src/main/java/com/hypherionmc/craterlib/nojang/server/BridgedMinecraftServer.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/nojang/server/BridgedMinecraftServer.java
@@ -9,6 +9,8 @@
 import lombok.RequiredArgsConstructor;
 import net.kyori.adventure.text.Component;
 import net.minecraft.SharedConstants;
+import net.minecraft.Util;
+import net.minecraft.network.chat.ChatType;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.players.UserBanListEntry;
 import net.minecraft.server.players.UserWhiteListEntry;
@@ -38,7 +40,7 @@
     }
 
     public String getName() {
-        return SharedConstants.getCurrentVersion().name();
+        return SharedConstants.getCurrentVersion().getName();
     }
 
     public boolean usesAuthentication() {
@@ -46,25 +48,25 @@
     }
 
     public void broadcastSystemMessage(Component text, boolean bl) {
-        internal.getPlayerList().broadcastSystemMessage(ChatUtils.adventureToMojang(text), bl);
+        internal.getPlayerList().broadcastMessage(ChatUtils.adventureToMojang(text), ChatType.SYSTEM, Util.NIL_UUID);
     }
 
     public boolean isPlayerBanned(BridgedGameProfile profile) {
-        return internal.getPlayerList().getBans().isBanned(profile.toNameAndId());
+        return internal.getPlayerList().getBans().isBanned(profile.toMojang());
     }
 
     public void whitelistPlayer(BridgedGameProfile gameProfile) {
         if (!internal.getPlayerList().isUsingWhitelist())
             return;
 
-        internal.getPlayerList().getWhiteList().add(new UserWhiteListEntry(gameProfile.toNameAndId()));
+        internal.getPlayerList().getWhiteList().add(new UserWhiteListEntry(gameProfile.toMojang()));
     }
 
     public void unWhitelistPlayer(BridgedGameProfile gameProfile) {
         if (!internal.getPlayerList().isUsingWhitelist())
             return;
 
-        internal.getPlayerList().getWhiteList().remove(new UserWhiteListEntry(gameProfile.toNameAndId()));
+        internal.getPlayerList().getWhiteList().remove(new UserWhiteListEntry(gameProfile.toMojang()));
     }
 
     public List<BridgedPlayer> getPlayers() {
@@ -83,11 +85,11 @@
     }
 
     public void banPlayer(BridgedGameProfile profile) {
-        internal.getPlayerList().getBans().add(new UserBanListEntry(profile.toNameAndId()));
+        internal.getPlayerList().getBans().add(new UserBanListEntry(profile.toMojang()));
     }
 
     public void executeCommand(BridgedMinecraftServer server, BridgedFakePlayer player, String command) {
-        internal.getCommands().performPrefixedCommand(player.toMojang(), command);
+        internal.getCommands().performCommand(player.toMojang(), command);
     }
 
     public MinecraftServer toMojang() {
