--- a/NeoForge/build.gradle
+++ b/NeoForge/build.gradle
@@ -51,7 +51,7 @@
     from project(":Common").sourceSets.main.resources
     def buildProps = project.properties.clone()
 
-    filesMatching("META-INF/neoforge.mods.toml") {
+    filesMatching("META-INF/mods.toml") {
         expand buildProps
     }
 }
@@ -106,8 +106,8 @@
     setVersionType("release")
     setChangelog("https://raw.githubusercontent.com/hypherionmc/changelogs/main/craterlib/changelog-forge.md")
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[NeoForge 1.21.2] CraterLib - ${project.version}")
-    setGameVersions("1.21.2")
+    setDisplayName("[NeoForge 1.20.4] CraterLib - ${project.version}")
+    setGameVersions("1.20.4")
     setLoaders("neoforge")
     setArtifact(remapJar)
     setCurseEnvironment("both")
