--- a/Common/src/main/java/com/hypherionmc/craterlib/mixin/events/PlayerListMixin.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/mixin/events/PlayerListMixin.java
@@ -12,7 +12,6 @@
 import net.minecraft.network.chat.Component;
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.server.network.CommonListenerCookie;
-import net.minecraft.server.players.NameAndId;
 import net.minecraft.server.players.PlayerList;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
@@ -50,9 +49,9 @@
     }
 
     @Inject(method = "canPlayerLogin", at = @At("HEAD"), cancellable = true)
-    private void injectPreLoginEvent(SocketAddress socketAddress, NameAndId arg, CallbackInfoReturnable<Component> cir) {
+    private void injectPreLoginEvent(SocketAddress address, GameProfile gameProfile, CallbackInfoReturnable<Component> cir) {
         try {
-            PlayerPreLoginEvent event = new PlayerPreLoginEvent(socketAddress, BridgedGameProfile.of(arg));
+            PlayerPreLoginEvent event = new PlayerPreLoginEvent(address, BridgedGameProfile.of(gameProfile));
             CraterEventBus.INSTANCE.postEvent(event);
             if (event.getMessage() != null) {
                 cir.setReturnValue(ChatUtils.adventureToMojang(event.getMessage()));
