--- a/Common/src/main/java/com/hypherionmc/craterlib/nojang/client/multiplayer/BridgedServerData.java
+++ b/Common/src/main/java/com/hypherionmc/craterlib/nojang/client/multiplayer/BridgedServerData.java
@@ -3,6 +3,7 @@
 import com.hypherionmc.craterlib.utils.ChatUtils;
 import lombok.RequiredArgsConstructor;
 import net.kyori.adventure.text.Component;
+import net.minecraft.ChatFormatting;
 import net.minecraft.client.multiplayer.ServerData;
 
 @RequiredArgsConstructor(staticName = "of")
@@ -23,11 +24,15 @@
     }
 
     public int getMaxPlayers() {
-        if (internal.players == null) {
+        if (!internal.pinged || internal.status.getString() == null) {
             return internal.playerList.size() + 1;
         }
 
-        return internal.players.max();
+        try {
+            return Integer.parseInt(ChatFormatting.stripFormatting(internal.status.getString()).split("/")[1]);
+        } catch (Exception ignored) {}
+
+        return 0;
     }
 
     public ServerData toMojang() {
