--- a/NeoForge/src/main/resources/META-INF/neoforge.mods.toml
+++ b/NeoForge/src/main/resources/META-INF/neoforge.mods.toml
@@ -16,22 +16,16 @@
     '''
     displayTest = "NONE"
 
-    [[mixins]]
-        config="${mod_id}.neoforge.mixins.json"
-
-    [[mixins]]
-        config="${mod_id}.mixins.json"
-
 [[dependencies.${ mod_id }]]
     modId = "neoforge"
     type="required"
-    versionRange = "[*,)"
+    versionRange = "[21.6.0-beta,)"
     ordering = "NONE"
     side = "BOTH"
 
 [[dependencies.${ mod_id }]]
     modId = "minecraft"
     type="required"
-    versionRange = "[*,)"
+    versionRange = "[1.21.6,)"
     ordering = "NONE"
     side = "BOTH"
