--- a/Paper/build.gradle
+++ b/Paper/build.gradle
@@ -68,8 +68,8 @@
     setVersionType("alpha")
     setChangelog(rootProject.file("changelog.md"))
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[Paper 1.21.9] CraterLib - ${project.version}")
-    setGameVersions("1.21.9")
+    setDisplayName("[Paper 1.21.6-8] CraterLib - ${project.version}")
+    setGameVersions("1.21.6", "1.21.7", "1.21.8")
     setLoaders("paper")
     setArtifact(reobfJar.outputJar)
 }
