--- a/.jenkins/Jenkinsfile.snapshot
+++ b/.jenkins/Jenkinsfile.snapshot
@@ -4,7 +4,7 @@
 def majorMc = "1.21";
 def modLoaders = "neoforge|fabric|quilt";
 def supportedMc = "1.21";
-def reltype = "port";
+def reltype = "snapshot";
 
 pipeline {
     agent {
@@ -32,15 +32,19 @@
 
         stage("Prepare") {
             steps {
-                sh "chmod +x ./gradlew"
-                sh "./gradlew build -PreleaseType=${reltype}"
+                dir("${WORKSPACE}/${majorMc}") {
+                    sh "chmod +x ./gradlew"
+                    sh "./gradlew build -PreleaseType=${reltype}"
+                }
             }
         }
 
         stage("Publish to Maven") {
             steps {
-                catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
-                    sh "./gradlew publish -PreleaseType=${reltype}"
+                dir ("${WORKSPACE}/${majorMc}") {
+                    catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
+                        sh "./gradlew publish -PreleaseType=${reltype}"
+                    }
                 }
             }
         }
@@ -48,20 +52,22 @@
 
     post {
         always {
-            sh "./gradlew --stop"
-            archiveArtifacts artifacts: 'artifacts/*.jar'
-
-            fddsnapshotter apiKey: env.PLATFORM_KEY,
-                projectSlug: "craterlib",
-                projectName: "${projectName}",
-                projectIcon: "${projectIcon}",
-                modLoaders: "${modLoaders}",
-                minecraftVersions: "${supportedMc}",
-                type: "snapshot",
-                failWebhook: env.SSS_WEBHOOK,
-                publishWebhooks: "${env.SSS_WEBHOOK}|${env.FDD_WH}"
-
-            deleteDir()
+            dir("${WORKSPACE}/${majorMc}") {
+                sh "./gradlew --stop"
+                archiveArtifacts artifacts: 'artifacts/*.jar'
+
+                fddsnapshotter apiKey: env.PLATFORM_KEY,
+                    projectSlug: "craterlib",
+                    projectName: "${projectName}",
+                    projectIcon: "${projectIcon}",
+                    modLoaders: "${modLoaders}",
+                    minecraftVersions: "${supportedMc}",
+                    type: "snapshot",
+                    failWebhook: env.SSS_WEBHOOK,
+                    publishWebhooks: "${env.SSS_WEBHOOK}|${env.FDD_WH}"
+
+                deleteDir()
+            }
         }
     }
 }
